src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php

```diff
@@ -194,4 +200,17 @@ private function mapRequestPayload(Request $request, string $type, MapRequestPay
             throw new HttpException(Response::HTTP_BAD_REQUEST, sprintf('Request payload contains invalid "%s" data.', $format), $e);
         }
     }
+
+    private function mapRequestHeader(Request $request, string $type, MapRequestHeader $attribute): ?object
+    {
+        $headers = [];
+        $requestHeaders = $request->headers->all();
+
+        array_walk($requestHeaders, function ($value, $key) use (&$headers) {
+            $key = lcfirst(str_replace('-', '', ucwords($key, '-')));
```

Comment:
Actually thinking further on this - I think assuming that the headers will always be kebab case might be a bit of a risk (I think headers usually always are but there doesn't seem to be a set rule on this that I can find) - IT might be better to just convert to camel from any format. 

I know symfony has this ability through the `String` component, but I am not sure if we can pull that into this resolver. You would need to check on that, maybe ask on the discussions or support channel? 

There example I found that does this with the package however was this:

`u($key)->camel()->toString();`

:)

Reply:
@robert-sykes 

The header will always be, if I'm not wrong, in kebab case because in the Resolver it's already formatted like this from the [HeaderBag](https://github.com/symfony/symfony/blob/9a0f178c5324c6ee7a3d99fe4fb7ef31b454925c/src/Symfony/Component/HttpFoundation/HeaderBag.php#L136) class when it set every key in the construct to return an array with keys in kebab case.

The only problem is,  if someone add a key in camelCase or PascalCase in his header it will be returned in lowercase without the - of separation.
But for me it's more about a problem of conception if someone do that I guess.

Reply:
@StevenRenaux right you are. That makes sense! Looks good though - this will be really handy and great to map headers to a class and validate. Thanks for putting in the work :)